rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -999,6 +999,69 @@ private Integer getProposedFederationFormatVersionFromRepository() {
+            assertFalse(federationStorageProvider.getProposedFederation(federationConstants, allActivations).isPresent());
+        }
+
+        @Test
+        void getProposedFederation_preRSKIP419_whenProposedFederationIsSet_shouldReturnEmpty() {
+            federationStorageProvider.setProposedFederation(proposedFederation);
+
+            assertFalse(federationStorageProvider.getProposedFederation(federationConstants, preLovellActivations).isPresent());
+        }
+
+        @Test
+        void getProposedFederation_whenProposedFederationIsSet_shouldReturnFederationSet() {
+            federationStorageProvider.setProposedFederation(proposedFederation);
+
+            assertEquals(Optional.of(proposedFederation), federationStorageProvider.getProposedFederation(federationConstants, allActivations));
+        }
+
+        @Test
+        void getProposedFederation_whenProposedFederationIsSetToNull_shouldReturnEmpty() {
+            federationStorageProvider.setProposedFederation(null);
+
+            assertFalse(federationStorageProvider.getProposedFederation(federationConstants, allActivations).isPresent());
+        }
+
+        @Test
+        void getProposedFederation_whenProposedFederationIsSaved_shouldReturnSavedFederation() {
+            // first we have to save the proposed fed so the repo is not empty
+            bridgeStorageAccessor.saveToRepository(PROPOSED_FEDERATION_FORMAT_VERSION.getKey(), proposedFederation.getFormatVersion(), BridgeSerializationUtils::serializeInteger);
+            bridgeStorageAccessor.saveToRepository(PROPOSED_FEDERATION.getKey(), proposedFederation, BridgeSerializationUtils::serializeFederation);
+
+            assertEquals(Optional.of(proposedFederation), federationStorageProvider.getProposedFederation(federationConstants, allActivations));
+        }
+
+        @Test
+        void getProposedFederation_whenNullProposedFederationIsSaved_shouldReturnEmpty() {
+            // first save a non-null value to make sure saving a null one is actually happening
+            bridgeStorageAccessor.saveToRepository(PROPOSED_FEDERATION_FORMAT_VERSION.getKey(), proposedFederation.getFormatVersion(), BridgeSerializationUtils::serializeInteger);
+            bridgeStorageAccessor.saveToRepository(PROPOSED_FEDERATION.getKey(), proposedFederation, BridgeSerializationUtils::serializeFederation);
+
+            bridgeStorageAccessor.saveToRepository(PROPOSED_FEDERATION_FORMAT_VERSION.getKey(), null, BridgeSerializationUtils::serializeInteger);
+            bridgeStorageAccessor.saveToRepository(PROPOSED_FEDERATION.getKey(), null, BridgeSerializationUtils::serializeFederation);
+
+            assertFalse(federationStorageProvider.getProposedFederation(federationConstants, allActivations).isPresent());
+        }
+
+        @Test
+        void getProposedFederation_whenProposedFederationIsCached_shouldReturnCachedFederation() {
```

Comment:
Nice, maybe another case when there is a fed in storage. Call `set`, not `save`, and call `get`. Should return the one setted and not the one in storage

Reply:
good one, thanks